DOCSP-26452: use struct in compound ops (#223)
mongodb/docs-golang: source/fundamentals/crud/compound-operations.txt

Example
```````

> The following example matches and deletes a document where the
> ``enrollment`` field value is less than 20 using the
> ``FindOneAndDelete()`` method:

Reviewer: Suggestion:

```suggestion
The following example uses the ``FindOneAndDelete()`` method 
to match and delete the first document where the``enrollment`` 
field value is less than 20:
```

Example
```````

> The following example performs the following actions in order using the
> ``FindOneAndUpdate()`` method:

Reviewer: Suggestion:

```suggestion
The following example uses the ``FindOneAndUpdate()`` method to 
perform the following actions in order:
```

> - Matches a document where the ``title`` field value includes "Modern"

Reviewer: Issue: If this is like the Find methods in other drivers, it specifically updates the first matching document, not just an arbitrary one. Probably good to make that clear.

Suggestion:

```suggestion
- Matches the first document where the ``title`` field value includes "Modern"
```

Example
```````

The following example performs the following actions in order using the

> ``FindOneAndReplace()`` method:

Reviewer: Suggest same change here as above

> - Matches a document where the ``title`` is "Representation Theory"

Reviewer: Same suggestion as above
